creusot/src/backend/term.rs

```diff
@@ -212,7 +216,12 @@ impl<'tcx, N: Namer<'tcx>> Lower<'_, 'tcx, N> {
             }
             TermKind::Old { box term } => Exp::Old(self.lower_term(term).boxed()),
             TermKind::Match { box scrutinee, arms } => {
-                if let PatternKind::Bool(b0) = arms[0].0.kind {
+                if arms.is_empty() {
+                    Exp::ascribe(
+                        Exp::qvar(self.names.in_pre(PreMod::Any, "any_l")).app([Exp::unit()]),
+                        translate_ty(self.ctx, self.names, term.span, term.ty),
+                    )
```

Comment:
The behavior you choose here is to use an indeterminate value when pattern matching on the ! type in Pearlite. Recall that in the logic, this type is interpreted as unit, because Why3 does not support empty types.

I would be rather in favor of simply not allowing empty pattern matching in Pearlite. Two reasons for this:

1- I don't see in which case pattern matching on ! this way can be useful. If the user needs an indeterminate value, then she can use dedicated features.

2- This is confusing, because this gives the wrong belief that ! is empty in Pearlite. I would greatly prefer a pedagogical error message.

Reply:
I see. I'm still not used to thinking that there are no empty types in Why3.